File: 9. Palindrome Number/solution.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
  if (x < 0 || (x % 10 === 0 && x !== 0)) return false;

  let originalNumber = x;
  let reverse = 0;

  while (x > 0) {
    reverse = reverse * 10 + (x % 10);
    x = Math.floor(x / 10);
  }

  return originalNumber - reverse === 0 ? true : false;
};

File: 9. Palindrome Number/solution.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """

        if x < 0 or (x % 10 == 0 and x != 0):
            return False
        
        originalNumber = x
        reverse = 0

        while x > 0:
            reverse = reverse * 10 + (x % 10)
            x = x // 10
        
        return True if originalNumber - reverse == 0 else False
        